### PR TITLE
Emails displayed in same thread.

### LINKED ISSUE #7194: Emails from same research note should display in same thread
Hi, this is a [first-timers-only issue](https://code.publiclab.org/#r=all). This means we've worked to make it more legible to folks who either **haven't contributed to our codebase before, or even folks who haven't contributed to open source before**. 

If that's you, we're interested in helping you take the first step and can answer questions and help you out as you do. Note that we're especially interested in contributions from people from groups underrepresented in free and open source software!

We know that the process of creating a pull request is the biggest barrier for new contributors. This issue is for you 💝

If you have contributed before, **consider leaving this one for someone new**, and looking through our general [help wanted](https://github.com/publiclab/plots2/labels/help-wanted) issues. Thanks!

### 🤔 What you will need to know.

Nothing. This issue is meant to welcome you to Open Source :) We are happy to walk you through the process.

### 📋 Step by Step

- [ ] 🙋 **Claim this issue**: Comment below. If someone else has claimed it, ask if they've opened a pull request already and if they're stuck -- maybe you can help them solve a problem or move it along!

- [ ] 📝 **Update** the file [$FILENAME]($BRANCH_URL) in the `$REPO` repository (press the little pen Icon) and edit the line as shown below. 

[See this page](https://code.publiclab.org/#r=all) for some help in taking your first steps!

### The issue

As noted in #6230, comments and like messages from multiple people on the same research note show up as separate threads in an email inbox. Instead, they should show up in the same thread. In order to achieve this, we have to make sure the titles of the emails match exactly.

### The fix

In the files `subscription_mailer.rb` and `comment_mailer.rb`, there are different methods for each type of event (e.g., when a note is updated, liked, or initially created. To make all the messages show up in the same email thread, we need to make all the subject lines look the same.

#### Changes to make in `subscription_mailer.rb`

Let's make all the subject lines look like this one, on line 7:
https://github.com/publiclab/plots2/blob/7e2ebf018d021f6f31d5a9d99d242eb5048ba2ba/app/mailers/subscription_mailer.rb#L7

First, replace the following three lines with the exact text from line 7 (above):
https://github.com/publiclab/plots2/blob/7e2ebf018d021f6f31d5a9d99d242eb5048ba2ba/app/mailers/subscription_mailer.rb#L22-L24

Then, in the `notify_tag_added` method, paste the exact text from line 7 right below the method header (i.e., right before the line saying `@tag = tag`):
https://github.com/publiclab/plots2/blob/7e2ebf018d021f6f31d5a9d99d242eb5048ba2ba/app/mailers/subscription_mailer.rb#L32-L59

Finally, on this line: 
https://github.com/publiclab/plots2/blob/7e2ebf018d021f6f31d5a9d99d242eb5048ba2ba/app/mailers/subscription_mailer.rb#L57
Replace `"#{node.title} (##{node.id})"` with `subject`.

#### Changes to make in `comment_mailer.rb`

Change line 11 of the file:
https://github.com/publiclab/plots2/blob/7e2ebf018d021f6f31d5a9d99d242eb5048ba2ba/app/mailers/comment_mailer.rb#L11

to match line 18:
https://github.com/publiclab/plots2/blob/7e2ebf018d021f6f31d5a9d99d242eb5048ba2ba/app/mailers/comment_mailer.rb#L18

- [ ] 💾 **Commit** your changes

- [ ] 🔀 **Start a Pull Request**. There are two ways how you can start a pull request:

1. If you are familiar with the terminal or would like to learn it, [here is a great tutorial](https://egghead.io/series/how-to-contribute-to-an-open-source-project-on-github) on how to send a pull request using the terminal.

2. You can also [edit files directly in your browser](https://help.github.com/articles/editing-files-in-your-repository/) and open a pull request from there. 

- [ ] 🏁 **Done** Ask in comments for a review :)

### Please keep us updated

💬⏰ - We encourage contributors to be respectful to the community and provide an update **within a week** of claiming a first-timers-only issue. We're happy to keep it assigned to you as long as you need if you update us with a request for more time or help, but if we don't see any activity a week after you claim it we may reassign it to give someone else a chance. Thank you in advance! 

If this happens to you, don't sweat it! Grab another open issue.

### Is someone else already working on this?

🔗- We encourage contributors to link to the original issue in their pull request so all users can easily see if someone's already started on it. 

👥- **If someone seems stuck, offer them some help!** Otherwise, [take a look at some other issues you can help with](https://code.publiclab.org/#r=all). Thanks!

### 🤔❓ Questions?

Leave a comment below!

### PR DESCRIPTION
**Pull request by anshalshukla**
Fixes #7194  (<=== Add issue number here)

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

* [x] PR is descriptively titled 📑 and links the original issue above 🔗
* [x] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [x] code is in uniquely-named feature branch and has no merge conflicts 📁
* [x] screenshots/GIFs are attached 📎 in case of UI updation
* [x] ask `@publiclab/reviewers` for help, in a comment below

> We're happy to help you get this ready -- don't be afraid to ask for help, and **don't be discouraged** if your tests fail at first!

If tests do fail, click on the red `X` to learn why by reading the logs.

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/contributing-to-public-lab-software 

Thanks!

### CODE REVIEW

**Comment by anshalshukla** [comment]
@anthony-zhou  please review this PR.

**Comment by jywarren** [comment]
I'm just restarting this! Thank you, i hope to review it today!!!

**Comment by jywarren** [comment]
Hmm. I guess we're a little stuck here! But I think it's possible the tests that confirm email titles are now failing... can you try looking in the test logs to see which failed and if they need to be adjusted to the new email titles? Thank you!

File: app/mailers/comment_mailer.rb [hunk 1]
```diff
@@ -10,8 +10,8 @@ def notify(user, comment)
     @user = user
     @comment = comment
     @footer = feature('email-footer')
-    mail(to: user.email, subject: "New comment on #{comment.parent.title}" \
-                                  " (##{comment.parent.id}) - #c#{comment.id} ")
+    mail(to: user.email, subject: "New comment on #{comment.node.title}" \
```
Comment (jywarren) [inline app/mailers/comment_mailer.rb:13]: Hi, I think `parent` is an alias for `node`, isn't it?

**Comment by jywarren** [comment]
Hi @anshalshukla -- it's been a while so I will close this, but if you'd like to re-open and keep working on it, you're welcome to, and we'd be grateful! Thanks!

**Pull request closed without merging**